base/cvd/cuttlefish/host/commands/assemble_cvd/android_build/combined_android_build.h

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright (C) 2025 The Android Open Source Project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "cuttlefish/common/libs/utils/result.h"
+#include "cuttlefish/host/commands/assemble_cvd/android_build/android_build.h"
+
+namespace cuttlefish {
+
+/**
+ * Combines information from multiple `AndroidBuild` instances.
+ *
+ * Individual sources can implement parts of the API, and an instance of
+ * this class combines non-error results to provide as much information as
+ * possible to the caller.
+ */
+Result<std::unique_ptr<AndroidBuild>> CombinedAndroidBuild(
```

Comment:
`CombinedAndroidBuild` imposes restrictions on the contents of the underlying `AndroidBuild`s because of the way it chooses what build to get a particular image from. For example, if the caller wants the system image from the second build it must ensure the first build returns error when asked for that image.

Would it make sense to create a `CombinedAndroidBuild` builder that would explicitly declare what build to use for each image? Something like this:
```
CombinedAndroidBuild::FromDefaultBuild(default_build).WithSystemImageFrom(system_build).WithSuperImageFrom(super_build)(...).Build();
```

Reply:
The intention is that a `CombinedAndroidBuild` is only ever made from components that are consistent with each other. If `assemble_cvd` is working with both a default build and a system build, those are intended to be represented by distinct `AndroidBuild`s.

I'm open to different names for this class. The list of implementations right now is `AndroidProductDir`, `FetchedAndroidBuild`, `ImgZip`, `MiscInfoMetadata`, `SuperImage`, and `TargetFiles`.